### PR TITLE
UI Service

### PR DESCRIPTION
**Pull request by aliok**
- A service that only serves the HTML+JS client
- Make `kvc`s report `http` in their `status.url` instead of `https`
- Enable CORS on UI service
- GitHub action to build and push the UI service
- Pass upload service URL to the UI service and thus to the HTML+JS client

### CODE REVIEW

**Comment by aliok** [comment]
cc @pierDipi

File: services/ui-service/templates/index.html [hunk 1]
```diff
@@ -121,25 +121,26 @@
...
-                    calc(-1*var(--d))     calc(0*var(--d))     0 8px,
-                    calc(-0.707*var(--d)) calc(-0.707*var(--d))0 10px,
-                    calc(0*var(--d))      calc(-1*var(--d))    0 12px;
+            box-shadow: calc(1 * var(--d)) calc(0 * var(--d)) 0 0,
+            calc(0.707 * var(--d)) calc(0.707 * var(--d)) 0 2px,
+            calc(0 * var(--d)) calc(1 * var(--d)) 0 4px,
+            calc(-0.707 * var(--d)) calc(0.707 * var(--d)) 0 6px,
+            calc(-1 * var(--d)) calc(0 * var(--d)) 0 8px,
+            calc(-0.707 * var(--d)) calc(-0.707 * var(--d)) 0 10px,
+            calc(0 * var(--d)) calc(-1 * var(--d)) 0 12px;
```
Comment (pierDipi) [inline services/ui-service/templates/index.html:135]: how do you make sense of this?

File: infra/openshift-manifests/lib.sh [hunk 2]
```diff
@@ -45,3 +45,26 @@ EOF
 function delete_minio_endpoint_route(){
     oc delete route -n minio-operator minio-endpoint
 }
+
+function patch_knative_serving(){
+    # patch knative serving to use http instead of https in the service status url
+    oc patch knativeserving -n knative-serving knative-serving -p '{"spec":{"config":{"network":{"default-external-scheme": "http"}}}}' --type=merge
+    # wait until knative serving is ready
+    oc wait --for=condition=Ready knativeserving -n knative-serving knative-serving
+}
```
Comment (pierDipi) [inline infra/openshift-manifests/lib.sh:54]: Ideally, we do this with kustomize

Comment (aliok) [reply infra/openshift-manifests/lib.sh:54]: Can you make that for me? I am busy with implementing the next round of services.
I don't know about Kustomize.

File: services/ui-service/templates/index.html (hunk 1 above)
Comment (aliok) [reply services/ui-service/templates/index.html:135]: I've used this tool: https://www.cssportal.com/css-loader-generator/

File: infra/openshift-manifests/lib.sh (hunk 2 above)
Comment (pierDipi) [reply infra/openshift-manifests/lib.sh:54]: it is not a blocker comment

**Review by pierDipi — APPROVED** [review]
LGTM

**Pull request merged**